CHANGELOG.md

```diff
@@ -4,6 +4,376 @@ All notable changes to Router will be documented in this file.
+
+### An APQ query with a mismatched hash will error as HTTP 400 ([Issue #2948](https://github.com/apollographql/router/issues/2948))
+
+We now model the both the behavior of the Gateway and the intended behavior of [the implementation](https://www.apollographql.com/docs/apollo-server/performance/apq/).  Even if our previous behavior was still acceptable, any other behavior is a misconfiguration of a client and should be prevented early.
+
+Previously, if a client sent an operation with an APQ hash, we would merely log an error to the console, **not** register the operation (for the next request) but still execute the query.  We now return a GraphQL error and don't execute the query.  No clients should be impacted by this, though anyone who had hand-crafted a query **with** APQ information (for example, copied a previous APQ-registration query but only changed the operation without re-calculating the SHA-256) might now be forced to use the correct hash (or more practically, remove the hash).
+
+By [@o0Ignition0o](https://github.com/o0Ignition0o) in https://github.com/apollographql/router/pull/3128
+
+### fix(subscription): take the callback url path from the configuration ([Issue #3361](https://github.com/apollographql/router/issues/3361))
+
+Previously when you specified the `subscription.mode.callback.path` it was not used, we had an hardcoded value set to `/callback`. It's now using the specified path in the configuration
+
+By [@bnjjj](https://github.com/bnjjj) in https://github.com/apollographql/router/pull/3366
+
+### Preserve all shutdown receivers across reloads ([Issue #3139](https://github.com/apollographql/router/issues/3139))
+
+Preserve a vec of all channels that we create and process all of them during shutdown. This will avoid scenarios such as:
+
+- some requests are in flight
+- the router reloads (new schema, etc)
+- the router gets a shutdown signal
+- since the shutdown channel for the older configuration is not kept, the router closes immediately without waiting for the initial connections to stop
+
+By [@garypen](https://github.com/garypen) in https://github.com/apollographql/router/pull/3311
+
+### Enable serde_json float_roundtrip feature ([Issue #2951](https://github.com/apollographql/router/issues/2951))
+
+This feature is explicitly designed to:
+
+"""
+Use sufficient precision when parsing fixed precision floats from JSON to ensure that they maintain accuracy when round-tripped through JSON. This comes at an approximately 2x performance cost for parsing floats compared to the default best-effort precision.
+"""
+
+Despite the performance impact for floats, we need the fix in order to prevent float values losing precision when processed by the router.
+
+By [@garypen](https://github.com/garypen) in https://github.com/apollographql/router/pull/3338
+
+### Fix @defer inside an aliased field ([Issue #3263](https://github.com/apollographql/router/issues/3263))
```

Comment:
maybe merge that entry with #3298 and #3339

Reply:
Agreed